app/actions/handlers.py

```diff
@@ -204,6 +191,18 @@ async def action_pull_events(integration, action_config: PullEventsConfig):
                             str(integration.id),
                             "pull_events"
                         )
+
+                        if response:
+                            # update states
+                            dataset_status.latest_updated_on = dataset_metadata.updated_on
+
+                            await state_manager.set_state(
+                                str(integration.id),
+                                "pull_events",
+                                dataset_status.json(),
+                                client.DATASET_GFW_INTEGRATED_ALERTS
+                            )
```

Comment:
Q: Looks like we use the state to set/check that timestamp only for Integrated Alerts, and not for VIIRS/Fire alerts right?
I don't remember how that was in v1, so it may be fine if it was intended.

Reply:
It is intended this way, Fire alerts are not using the state manager